src/dataAPI/jupyter/PythonExecutor.ts

```diff
@@ -267,6 +269,87 @@ export class PythonPandasExecutor {
         }
     }
 
+    public async textFormat(
+        dfName: string,
+        colName: string,
+        type:string,
```

Comment:
It seems like you're using type to differentiate what kind of fact to generate. There are technically 5 main types right now in `pandas-data-types.ts` but the main ones we care about are numeric boolean and categorical. Can you use the full words and do a union type so that it is `type: "numeric" | "categorical" | timestamp"` or something